core/container/src/main/java/org/wildfly/swarm/container/runtime/cdi/TempFileProviderProducer.java

```diff
@@ -61,14 +61,12 @@ TempFileProvider tempFileProvider() {
         return this.tempFileProvider;
     }
 
-    void dispose(@Disposes TempFileProvider provider) {
-        // To ensure we only close the one we produce
-        if (this.tempFileProvider == provider) {
-            try {
-                this.tempFileProvider.close();
-            } catch (IOException e) {
-                SwarmMessages.MESSAGES.errorCleaningUpTempFileProvider(e);
-            }
+    @PreDestroy
```

Comment:
This was curious to find. If having a non-static Dispose method, then class is constructed again, causing to this class to initialize again before Weld being shutdown (so 2 initializes, 1 destroy), hence having two temporal files, only 1 is removed.
Adding PreDestroy makes unloading safer and working.